### PR TITLE
docs: added contributing guide

### PR DESCRIPTION
**Pull request by anikethsaha**
Added contributing guide for contribution

**PR Type**
Docs

**Added test for the change , if required ?**
Not required

**Any breaking changes**
Nope

**Update the documentation ?**
NA

**Summary**
Added contributing guide for contribution

**Also**
I was thinking to add the above as PR template.

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# Contributing
+This page contains the contributing guidelines.
+
+## Setup
+### prerequisite
+- Node.js installed.
```
Comment (kenotron) [inline .github/CONTRIBUTING.md]: I think I might have used code that requires at least Node 8. I would recommend people to use the LTS version for development at this point.

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
+# Contributing
+This page contains the contributing guidelines.
+
+## Setup
+### prerequisite
+- Node.js installed. 
+ _Note: Node 6 or greater would be better for "best results"._
+- Package manager - [yarn](https://yarnpkg.com/en/docs/getting-started) or [npm](https://docs.npmjs.com/)
```
Comment (kenotron) [inline .github/CONTRIBUTING.md]: I am using Lerna + Yarn workspaces in this repo, so yarn is required.

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+
+
+### Project Setup in your local machine
+-   Fork the **just** repo at [https://github.com/microsoft/just](https://github.com/microsoft/just).
+-   `git clone <your-clone-url> && cd just`
+
+### Dependency Installation
+
+- Using npm
```
Comment (kenotron) [inline .github/CONTRIBUTING.md]: This section about npm should probably not be there

File: .github/CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+$ npm run build
+$ npm link
+```
+
+
+- Using yarn
+```bash
+$ yarn  
+$ yarn build
+$ yarn link
```
Comment (kenotron) [inline .github/CONTRIBUTING.md]: Don't think we need yarn link here. Maybe inside your for local testing?

File: .github/CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+```
+
+### Running and Testing locally
+
+**To run this with node**
+- Using npm
```
Comment (kenotron) [inline .github/CONTRIBUTING.md]: again, we don't use npm here, even though this works

File: .github/CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+$ npm run start
+```
+- Using yarn
+```bash
+$ yarn start
+```
+
+**To test the project** 
+
+- Using npm
```
Comment (kenotron) [inline .github/CONTRIBUTING.md]: test with yarn test

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (anikethsaha) [reply .github/CONTRIBUTING.md]: v8.0.0 it is then ?

File: .github/CONTRIBUTING.md (hunk 2 above)
Comment (anikethsaha) [reply .github/CONTRIBUTING.md]: yeah I was thinking to make that one as * preferred*.
Okay will change it

File: .github/CONTRIBUTING.md (hunk 4 above)
Comment (anikethsaha) [reply .github/CONTRIBUTING.md]: actually yup local testing only I kept that. Should I keep it ?

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (kenotron) [reply .github/CONTRIBUTING.md]: Sure!

File: .github/CONTRIBUTING.md (hunk 4 above)
Comment (kenotron) [reply .github/CONTRIBUTING.md]: You should remove it for now. I'm not sure how to best to word the local testing yet :) I've been using lots of yarn linking manually, etc. so far.

File: .github/CONTRIBUTING.md (hunk 3 above)
Comment (anikethsaha) [reply .github/CONTRIBUTING.md]: done

File: .github/CONTRIBUTING.md (hunk 2 above)
Comment (anikethsaha) [reply .github/CONTRIBUTING.md]: done 👍

**Comment by anikethsaha** [comment]
Also, I was thinking will it be good if we can add commitlint and commit hook using `husky` to check the format for the commit subject. These will help in generating a better changelog. 
cc @kenotron

**Comment by kenotron** [comment]
> Also, I was thinking will it be good if we can add commitlint and commit hook using `husky` to check the format for the commit subject. These will help in generating a better changelog.
> cc @kenotron Ken Chau FTE

This repo uses change files - not commits for changelog generation. So that needs to be documented as well (needs to run `yarn change`) at root to make this work.

**Comment by anikethsaha** [comment]
So I need to add those in this PR too? Actually, this was documentation related PR so I made that from GitHub itself, not from my local git. I dont have this branch on my pc. if I pull this then I guess I can get those changes

**Comment by kenotron** [comment]
No need to add those... not sure why the yarn check is failing, gonna look at it right now

**Comment by anikethsaha** [comment]
Ok then !...yeah the whole build is failing for that

**Comment by kenotron** [comment]
just fixed master - will merge this asap

**Comment by anikethsaha** [comment]
Awesome!  👍

**Pull request merged**